spras/analysis/summary.py

```diff
@@ -50,8 +51,44 @@ def summarize_networks(file_paths: Iterable[Path], node_table: pd.DataFrame, alg
         number_edges = nw.number_of_edges()
         ncc = nx.number_connected_components(nw)
 
+        # Save the max/median degree, average clustering coeffienct, and density
+        if number_nodes <= 1 or number_edges == 0:
+            max_degree = 0
+            median_degree = 0.0
+            density = 0.0
+        else:
+            degrees = [deg for _, deg in nw.degree()]
+            max_degree = max(degrees)
+            median_degree = median(degrees)
+            density = nx.density(nw)
+
+        # Save the max diameter
+        cc = list(nx.connected_components(nw))
+        diameters = [
+            nx.diameter(nw.subgraph(c).copy()) if len(c) > 1 else 0
+            for c in cc
+        ]
+        max_diameter = max(diameters, default=0)
+
+        # # Save the max average clustering coefficient
```

Comment:
This will only work for one component. With multiple the value is 0.0 always.

Reply:
We can drop this commented code then.